code/iaas/lb-logic/src/main/java/io/cattle/iaas/lb/api/action/LoadBalancerConfigSetListenersActionHandler.java

```diff
@@ -83,9 +74,7 @@ private void removeOldListenerMaps(LoadBalancerConfig config, List<? extends Lon
         }
 
         for (LoadBalancerConfigListenerMap mapToRemove : mapsToRemove) {
-            objectProcessManager.scheduleProcessInstance(LoadBalancerConstants.PROCESS_LB_CONFIG_LISTENER_MAP_REMOVE,
-                    mapToRemove, null);
+            lbService.removeListenerFromConfig(config, mapToRemove.getLoadBalancerListenerId());
```

Comment:
Probably not a big deal, but with the new changes, we now do a DB lookup for each map since we're just passing the ID here.  I didn't check but I'm guessing nothing is locked so I don't know if we get much out of the findNonRemoved call in lbService.removeListenerFromConfig()